### PR TITLE
Add URL public key option to `verify`, consolidate all options into a single flag

### PR DESCRIPTION
**Pull request by dekkagaijin**
Fixes https://github.com/sigstore/cosign/issues/247

Signed-off-by: Jake Sanders <jsand@google.com>

### CODE REVIEW

**Comment by dekkagaijin** [comment]
```
$ ./cosign verify -key-url https://raw.githubusercontent.com/GoogleContainerTools/distroless/main/cosign.pub gcr.io/distroless/base

Verification for gcr.io/distroless/base --
The following checks were performed on each of these signatures:
  - The cosign claims were validated
  - The signatures were verified against the specified public key
  - Any certificates were verified against the Fulcio roots.

[{"critical":{"identity":{"docker-reference":""},"image":{"docker-manifest-digest":"sha256:6ec6da1888b18dd971802c2a58a76a7702902b4c9c1be28f38e75e871cedc2df"},"type":"cosign container signature"},"optional":null},{"critical":{"identity":{"docker-reference":""},"image":{"docker-manifest-digest":"sha256:6ec6da1888b18dd971802c2a58a76a7702902b4c9c1be28f38e75e871cedc2df"},"type":"cosign container signature"},"optional":null},{"critical":{"identity":{"docker-reference":""},"image":{"docker-manifest-digest":"sha256:6ec6da1888b18dd971802c2a58a76a7702902b4c9c1be28f38e75e871cedc2df"},"type":"cosign container signature"},"optional":null}]
```
:)

File: cmd/cosign/cli/verify.go [hunk 1]
```diff
@@ -87,17 +92,20 @@ func (c *VerifyCommand) Exec(ctx context.Context, args []string) error {
 	if len(args) == 0 {
 		return flag.ErrHelp
 	}
-	if !oneOf(c.Key, c.KmsVal) {
+	if !oneOf(c.Key, c.KeyURL, c.KmsVal) {
```
Comment (dlorenc) [inline cmd/cosign/cli/verify.go:95]: wdyt about trying to handle this transparently in the same flag, like kubectl does?

Comment (dekkagaijin) [reply cmd/cosign/cli/verify.go:95]: works for me

Comment (dekkagaijin) [reply cmd/cosign/cli/verify.go]: done

File: cmd/cosign/cli/verify.go [hunk 2]
```diff
@@ -34,10 +34,9 @@ import (
 // VerifyCommand verifies a signature on a supplied container image
 type VerifyCommand struct {
 	CheckClaims bool
-	KmsVal      string
-	Key         string
+	KeyRef      string
 	Output      string
-	Annotations *map[string]interface{}
+	Annotations map[string]interface{}
```
[the remark this answers is not in the document]
Comment (dekkagaijin) [reply cmd/cosign/cli/verify.go]: done

**Review by dlorenc — APPROVED** [review]
This is amazing. Nicely done.

**Pull request merged**